File: cfgs/pythoncfgs/CHECK_GIVEN_STRING_CAN_SPLIT_FOUR_DISTINCT_STRINGS.py
Nodes
1;conditional;fun_1(foo_1) >= 10;8;
2;conditional;foo < foo;10;
3;statement;return false;19;
4;conditional;foo < foo;11;
5;conditional;foo < foo;12;
6;statement;foo_1 = foo_2[0:foo_3];13;
7;statement;foo_1 = foo_2[foo_3:foo_4 - foo_5];14;
8;statement;foo_1 = foo_2[foo_3:foo_4 - foo_5];15;
9;statement;foo_1 = foo_2[foo_3:fun_1(foo_4) - foo_5];16;
10;conditional;foo_1 != foo_2 and foo_3 != foo_4 and (foo_5 != foo_6) and (foo_7 != foo_8) and (foo_9 != foo_10) and (foo_11 != foo_12);17;
11;statement;return true;18;
12;statement;return true;9;
13;exit;;
Edges
0;2;12;
1;3;4;
2;13;
3;2;5;
4;4;6;
5;7;
6;8;
7;9;
8;10;
9;5;11;
10;13;
11;13;
12;

File: cfgs/java_cfgs_foo/CHECK_GIVEN_STRING_CAN_SPLIT_FOUR_DISTINCT_STRINGS.java
Nodes
1;special;;;
2;statement;int seg_len = fun_1() - foo_2;13;
3;statement;int res = 0;15;
4;statement;int i = 0;17;
5;conditional;foo_1 < foo_2;17;
6;statement;res = (foo_1 * 10) + (fun_1(foo_3) - '0');18;
7;statement;i++;17;
8;statement;int seg_len_pow = ((int) (fun_1(10, foo_1 - 1)));20;
9;statement;int curr_val = foo_1;22;
10;statement;int i = 1;24;
11;conditional;foo_1 <= (fun_1() - foo_3);25;
12;statement;curr_val = foo_1 - ((fun_1(foo_3 - 1) - '0') * foo_4);27;
13;statement;curr_val = (foo_1 * 10) + (fun_1((foo_3 + foo_4) - 1) - '0');30;
14;statement;res = fun_1(foo_1, foo_2);33;
15;statement;i++;25;
16;statement;return foo_1;36;
17;exit;;;
Edges
0;2;
1;3;
2;4;
3;5;
4;6;8;
5;7;
6;5;
7;9;
8;10;
9;11;
10;12;16;
11;13;
12;14;
13;15;
14;11;
15;17;
16;
